src/dimension.c

```diff
@@ -1680,7 +1678,7 @@ ts_dimension_add(PG_FUNCTION_ARGS)
 	TS_PREVENT_FUNC_IF_READ_ONLY();
 
 	if (!PG_ARGISNULL(1))
-		memcpy(&info.colname, PG_GETARG_NAME(1), NAMEDATALEN);
+		namestrcpy(&info.colname, PG_GETARG_NAME(1)->data);
```

Comment:
```suggestion
		namestrcpy(&info.colname, NameStr(PG_GETARG_NAME(1)));
```

Reply:
I was also experimenting with this, but I think it is harder to understand since you also need to dereference the result of `PG_GETARG_NAME`. But I changed it to be more consistent.